src/components/App.js

```diff
@@ -1,38 +1,31 @@
 import React from 'react';
 import Calculator from './Calculator';
+import calculate from '../logic/calculate';
 
 export default class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      buttons: [
-        { content: 'AC', className: 'btn' },
-        { content: '+/-', className: 'btn' },
-        { content: '%', className: 'btn' },
-        { content: '/', className: 'btn btn-orange' },
-        { content: '7', className: 'btn' },
-        { content: '8', className: 'btn' },
-        { content: '9', className: 'btn' },
-        { content: 'X', className: 'btn btn-orange' },
-        { content: '4', className: 'btn ' },
-        { content: '5', className: 'btn' },
-        { content: '6', className: 'btn' },
-        { content: '-', className: 'btn btn-orange' },
-        { content: '1', className: 'btn' },
-        { content: '2', className: 'btn' },
-        { content: '3', className: 'btn' },
-        { content: '+', className: 'btn btn-orange' },
-        { content: '0', className: 'btn' },
-        { content: '.', className: 'btn' },
-        { content: '=', className: 'btn btn-orange' },
-      ],
+      total: null,
+      next: null,
+      operation: null,
     };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(btnName) {
+    const { total, next, operation } = this.state;
+    this.setState(() => calculate({ total, next, operation }, btnName));
   }
 
   render() {
-    const { buttons } = this.state;
+    const { next, total } = this.state;
     return (
-      <Calculator buttons={buttons} />
+      <Calculator
+        click={this.handleClick}
+        display={next || total || '0'}
```

Comment:
- [ ] There is a bug in your display prop method. When I subtract a bigger number from a number (e.g:- 4 - 9), the result is not displayed as it should be. Kindly see the attached gif below.
![chrome-capture (2)](https://user-images.githubusercontent.com/64894927/129421180-cd5f7a92-219f-4dcf-882f-77ea28d2bc57.gif)